src/lib/core/media-marshaller/media-marshaller.ts

```diff
@@ -294,11 +312,15 @@ export class MediaMarshaller {
    * Watch for mediaQuery breakpoint activations
    */
   private observeActivations() {
+    const target = this as unknown as HookTarget;
     const queries = this.breakpoints.items.map(bp => bp.mediaQuery);
+
     this.matchMedia
-        .observe(queries)
-        .subscribe(this.activate.bind(this));
+        .observe(this.hook.withPrintQuery(queries))
+        .pipe(filter(this.hook.interceptEvents(target)))
```

Comment:
I still think this needs to be revisited, because it's not really filtering. Can we do a combination of `tap` and `filter`?

Reply:
Would look something like:
`tap(this.hook.interceptEvents(target), filter(this.hook.canActivate)`

Reply:
This is  filtering all `print ...` events. `filter(...)` is  the appropriate operator.